.scalafmt.conf

```diff
@@ -0,0 +1,20 @@
+maxColumn = 112
```

Comment:
The origin of this standard is not relevant for this discussion.
  
> Changing our code base so someone can read our code in mobile phones make no sense.
  
You missed the laptop part. Why are you changing the codebase at all then? Is this a real attempt to optimize for the contributors of sbt? Lots of people read code in their mobile phones. You don't think this is true? Have a look at the [Scalajs style guide](https://github.com/scala-js/scala-js/blob/master/CODINGSTYLE.md), particularly at the line length section. Alike them, I *do review* code in my mobile phone.
  
> Why not 132 columns (number of characters printable by IBM 1403) or 100 columns (Google's Java Guide)?
  
Google sticks to 80 in almost all the other programming languages they use. See, for instance, [Google's Python style guide](https://google.github.io/styleguide/pyguide.html#Line_length). The reason to go for 100 for Java was probably because Java is a very verbose language.
  
These are the strongest reasons why we should stick to 80 columns:
  
* Readable with a larger font on laptops. Sometimes it's a matter of taste, some other times it's an imperative to use larger fonts, especifically for people with eye problems (like me).
* Limits the complexity in one line. Java is a *very verbose* programming language. Scala, however, is a very complex and concise language that allows the programmer to increase the complexity of the logic in one line. Having shorter lines reduces the complexity of method bodies significantly and allows the programmer to think more about the code he's writing. Having a very long line is usually a code smell that hints that the code should be refactored.
  
Some other arguments in favor:
* Leaves space for putting two versions side by side for comparison. Lots of times, we diff and blame stuff and having long lines will make it an uncomfortable experience.
* Leaves space for navigation views in the IDE and other windows in the same screen.
* Limits indentation, which in turn limits complexity of methods/functions.
  
Another data point: designers use a column size between 50 and 80 in websites, more than that is considered bad for readability. There are usability studies that back this up.
  
As with standards, one should always have good reasons to go against them because there are always good reasons for them to become standard.